### PR TITLE
fix(wordcloud,choropleth): provide corrected names, deprecate old names

### PR DESCRIPTION
**Pull request by nstuyvesant**
### Updates

- In core and dependent packages, renames WorldCloudOptions to WordCloudOptions (a previous typo), provides alias for original name and marks alias as deprecated.
-  In core and dependent packages, renames ExperimentalChoroplethChart to ChoroplethChart, provides alias for original name and marks alias as deprecated.
- Removed superfluous typing for data property per Angular component as base already established the type (lint).
- Removed experimental from demos
- Removed double initialization of required properties for Svelte

### CODE REVIEW

File: packages/vue/src/index.ts [hunk 1]
```diff
@@ -128,8 +136,8 @@ export type {
 	StackedAreaChartOptions,
 	TreeChartOptions,
 	TreemapChartOptions,
-	WorldCloudChartOptions
```
Comment (theiliad) [inline packages/vue/src/index.ts:131]: we should export both and mark world as deprecated

File: packages/vue/src/components/CcvWordCloudChart.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { WordCloudChart, type WorldCloudChartOptions } from '@carbon/charts'
+import { WordCloudChart, type WordCloudChartOptions } from '@carbon/charts'
 import { chartFactory } from './chartFactory'
 
-export default chartFactory<WorldCloudChartOptions>(WordCloudChart, 'CcvWordCloudChart')
+export default chartFactory<WordCloudChartOptions>(WordCloudChart, 'CcvWordCloudChart')
```
Comment (theiliad) [inline packages/vue/src/components/CcvWordCloudChart.ts:4]: we should still export it as world as well and mark as deprecated

File: packages/vue/src/components/CcvChoroplethChart.ts [hunk 3]
```diff
@@ -1,7 +1,4 @@
-import {
-	ExperimentalChoroplethChart as ChoroplethChart,
-	type ChoroplethChartOptions
-} from '@carbon/charts'
+import { ChoroplethChart, type ChoroplethChartOptions } from '@carbon/charts'
 import { chartFactory } from './chartFactory'
 
 export default chartFactory<ChoroplethChartOptions>(
```
Comment (theiliad) [inline packages/vue/src/components/CcvChoroplethChart.ts:4]: we should still export ExperimentalChoroplethChart as well and mark as deprecated

File: packages/svelte/src/lib/ChoroplethChart.svelte [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <script lang="ts">
 	import {
-		ExperimentalChoroplethChart as ChoroplethChartCore,
+		ChoroplethChart as ChoroplethChartCore,
```
Comment (theiliad) [inline packages/svelte/src/lib/ChoroplethChart.svelte:3]: we should still export both and mark as deprecated

File: packages/react/src/index.ts [hunk 5]
```diff
@@ -69,8 +73,8 @@ export type {
 	StackedAreaChartOptions,
 	TreeChartOptions,
 	TreemapChartOptions,
-	WorldCloudChartOptions
+	WordCloudChartOptions
```
Comment (theiliad) [inline packages/react/src/index.ts:76]: we should still export both and mark as deprecated

File: packages/svelte/src/lib/WordCloudChart.svelte [hunk 6]
```diff
@@ -1,12 +1,12 @@
...
 	import {
 		WordCloudChart as WordCloudChartCore,
-		type WorldCloudChartOptions,
+		type WordCloudChartOptions,
 		type ChartTabularData
 	} from '@carbon/charts'
 	import BaseChart from './BaseChart.svelte'
 
-	export let options: WorldCloudChartOptions
+	export let options: WordCloudChartOptions
```
Comment (theiliad) [inline packages/svelte/src/lib/WordCloudChart.svelte:9]: we should still use both

File: packages/react/src/charts/WordCloudChart.tsx [hunk 7]
```diff
@@ -1,12 +1,12 @@
...
 	WordCloudChart as WordCloudChartCore,
-	type WorldCloudChartOptions,
+	type WordCloudChartOptions,
 	type ChartTabularData
 } from '@carbon/charts'
 import BaseChart from './BaseChart'
 
-export default class WordCloudChart extends BaseChart<WorldCloudChartOptions> {
-	createChart(chartRef: HTMLDivElement, data: ChartTabularData, options: WorldCloudChartOptions) {
+export default class WordCloudChart extends BaseChart<WordCloudChartOptions> {
```
Comment (theiliad) [inline packages/react/src/charts/WordCloudChart.tsx:8]: we should still use both

**Review by theiliad — CHANGES_REQUESTED** [review]

File: packages/svelte/src/lib/ChoroplethChart.svelte (hunk 4 above)
Comment (nstuyvesant) [reply packages/svelte/src/lib/ChoroplethChart.svelte:3]: The export in packages/svelte/src/lib/index.ts exports both and marks ExperimentalChoroplethChart as deprecated.

File: packages/vue/src/components/CcvChoroplethChart.ts (hunk 3 above)
Comment (nstuyvesant) [reply packages/vue/src/components/CcvChoroplethChart.ts:4]: The export is in packages/vue/src/index.ts. Both are exported with the experimental one marked deprecated.

File: packages/vue/src/index.ts (hunk 1 above)
Comment (nstuyvesant) [reply packages/vue/src/index.ts:131]: Updated

File: packages/vue/src/components/CcvWordCloudChart.ts (hunk 2 above)
Comment (nstuyvesant) [reply packages/vue/src/components/CcvWordCloudChart.ts]: Updated

File: packages/react/src/index.ts (hunk 5 above)
Comment (nstuyvesant) [reply packages/react/src/index.ts]: Updated

File: packages/react/src/charts/WordCloudChart.tsx (hunk 7 above)
Comment (nstuyvesant) [reply packages/react/src/charts/WordCloudChart.tsx]: Revised and pushed.

File: packages/svelte/src/lib/WordCloudChart.svelte (hunk 6 above)
Comment (nstuyvesant) [reply packages/svelte/src/lib/WordCloudChart.svelte]: Revised and pushed.

File: packages/core/src/charts/wordcloud.ts [hunk 8]
```diff
@@ -12,7 +12,7 @@ import { WordCloud } from '@/components/graphs/wordcloud'
 export class WordCloudChart extends Chart {
 	model = new WordCloudModel(this.services)
 
-	constructor(holder: HTMLDivElement, chartConfigs: ChartConfig<WorldCloudChartOptions>) {
+	constructor(holder: HTMLDivElement, chartConfigs: ChartConfig<WordCloudChartOptions>) {
```
Comment (theiliad) [inline packages/core/src/charts/wordcloud.ts:15]: should we do `WordCloudChartOptions | WorldCloudChartOptions`? typescript could still throw an error not matching the type names

File: packages/core/src/interfaces/charts.ts [hunk 9]
```diff
@@ -627,4 +627,4 @@ export type ChartOptions =
 	| StackedBarChartOptions
 	| TreeChartOptions
 	| TreemapChartOptions
-	| WorldCloudChartOptions
+	| WordCloudChartOptions
```
Comment (theiliad) [inline packages/core/src/interfaces/charts.ts:630]: should we do `WordCloudChartOptions | WorldCloudChartOptions`? typescript could still throw an error not matching the type names

File: packages/angular/src/lib/charts/experimental-choropleth-chart-component.ts [hunk 10]
```diff
@@ -0,0 +1,29 @@
+import { Component, AfterViewInit } from '@angular/core'
+import { BaseChartComponent } from './base-chart.component'
+import { ChoroplethChart as ChoroplethChartCore, type ChoroplethChartOptions } from '@carbon/charts'
+
+/**
+ * Wrapper around `ChoroplethChart` in carbon charts library for deprecated selector.
+ *
+ * Most functions just call their equivalent from the chart library.
+ */
+@Component({
```
Comment (theiliad) [inline packages/angular/src/lib/charts/experimental-choropleth-chart-component.ts:10]: is there a way in angular to mark this as deprecated?

File: packages/angular/src/lib/charts/wordcloud-chart.component.ts [hunk 11]
```diff
@@ -20,9 +16,9 @@ export class WordCloudChartComponent extends BaseChartComponent implements After
 	 * Runs after view init to create a chart, attach it to `elementRef` and draw it.
 	 */
 	override ngAfterViewInit() {
-		this.chart = new WordCloudChartCore(this.elementRef.nativeElement, {
-			data: this.data as ChartTabularData,
-			options: this.options as WorldCloudChartOptions
+		this.chart = new WordCloudChartCore(this.elementRef.nativeElement as HTMLDivElement, {
+			data: this.data,
+			options: this.options as WordCloudChartOptions
```
Comment (theiliad) [inline packages/angular/src/lib/charts/wordcloud-chart.component.ts:21]: should we do WordCloudChartOptions | WorldCloudChartOptions? typescript could still throw an error not matching the type names

**Review by theiliad — CHANGES_REQUESTED** [review]

File: packages/angular/src/lib/charts/experimental-choropleth-chart-component.ts (hunk 10 above)
Comment (nstuyvesant) [reply packages/angular/src/lib/charts/experimental-choropleth-chart-component.ts:11]: Added JSdoc `@deprecated` to the component but if you check packages/angular/src/index.ts, you'll see that the export flags it as deprecated there as well.

File: packages/angular/src/lib/charts/wordcloud-chart.component.ts (hunk 11 above)
Comment (nstuyvesant) [reply packages/angular/src/lib/charts/wordcloud-chart.component.ts]: Got this updated.

File: packages/core/src/interfaces/charts.ts (hunk 9 above)
Comment (nstuyvesant) [reply packages/core/src/interfaces/charts.ts:630]: Both are exported in packages/core/src/interfaces/index.ts which is the export point for the interfaces namespace. Does that address this concern?

File: packages/core/src/charts/wordcloud.ts (hunk 8 above)
Comment (nstuyvesant) [reply packages/core/src/charts/wordcloud.ts]: Updated.

File: packages/core/src/interfaces/charts.ts (hunk 9 above)
Comment (nstuyvesant) [reply packages/core/src/interfaces/charts.ts:632]: Updated.

**Review by theiliad — APPROVED** [review]

**Pull request merged**